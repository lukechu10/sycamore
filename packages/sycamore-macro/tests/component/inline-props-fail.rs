use sycamore::prelude::*;

#[component(not_inline_props)]
fn NotInlineProps() -> View {
    view! {}
}

#[component(inline_props)]
fn ReceiverProp(self) -> View {
    view! {}
}

struct Foo {
    bar: i32,
}

#[component(inline_props)]
fn PatternWithoutIdent(Foo { bar }: Foo) -> View {
    view! {
        (bar)
    }
}

fn main() {}
