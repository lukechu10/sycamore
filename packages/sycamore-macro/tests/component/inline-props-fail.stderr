error: expected `inline_props`
 --> tests/component/inline-props-fail.rs:3:13
  |
3 | #[component(not_inline_props)]
  |             ^^^^^^^^^^^^^^^^

error: `self` cannot be a property
 --> tests/component/inline-props-fail.rs:9:17
  |
9 | fn ReceiverProp(self) -> View {
  |                 ^^^^

error: pattern must contain an identifier, properties cannot be unnamed
  --> tests/component/inline-props-fail.rs:18:24
   |
18 | fn PatternWithoutIdent(Foo { bar }: Foo) -> View {
   |                        ^^^

error: `self` parameter is only allowed in associated functions
 --> tests/component/inline-props-fail.rs:9:17
  |
9 | fn ReceiverProp(self) -> View {
  |                 ^^^^ not semantically valid as function parameter
  |
  = note: associated functions are those in `impl` or `trait` definitions
